tests/lib/rules/valid-jsdoc.js

```diff
@@ -287,6 +287,22 @@ eslintTester.addRuleTest("lib/rules/valid-jsdoc", {
                 message: "Unexpected @returns tag; function has no return statement.",
                 type: "Block"
             }]
+        },
+        {
+            code: "/**\n * Some text\n *\n * @param arg [optional] text description\n */\nfunction testFunction(arg) {}",
```

Comment:
Re-validating that my fix does what it's supposed to I've removed the `tag.name` check from the if-clause. 

This is the result from running Makefile.js test: 
```
$> node Makefile.js test                                                                                     
Validating Makefile.js                                                                                                                                                 
Validating JSON Files                                                                                                                                                  
Validating Markdown Files                                                                                                                                              
Validating JavaScript files                                                                                                                                            
Validating JavaScript test files                                                                                                                                       
Validating rules                                                                                                                                                       
                                                                                                                                                                       
  [▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬▬]                                                                                                                               
                                                                                                                                                                       
  5217 passing (9s)                                                                                                                                                    
  1 failing                                                                                                                                                            
                                                                                                                                                                       
  1) valid-jsdoc /**                                                                                                                                                   
 * Some text                                                                                                                                                           
 *                                                                                                                                                                     
 * @param arg [optional] text description                                                                                                                              
 */                                                                                                                                                                    
function testFunction(arg) {}:                                                                                                                                         
     TypeError: Cannot read property 'indexOf' of undefined                                                                                                            
      at lib/rules/valid-jsdoc.js:9:3535                                                                                                                               
      at Array.forEach (native)                                                                                                                                        
      at EventEmitter.checkJSDoc (lib/rules/valid-jsdoc.js:9:2369)                                                                                                     
      at Controller.controller.traverse.leave (lib/eslint.js:9:29079)                                                                                                  
      at EventEmitter.module.exports.api.verify (lib/eslint.js:9:28389)                                                                                                
                                                                                                                                                                       
                                                                                                                                                                       
                                                                                                                                                                       
=============================================================================                                                                                          
Writing coverage object [/Users/amarschke/src/js/eslint-fix-jsdoc/coverage/coverage.json]                                                                              
Writing coverage reports at [/Users/amarschke/src/js/eslint-fix-jsdoc/coverage]                                                                                        
=============================================================================                                                                                          
                                                                                                                                                                       
=============================== Coverage summary ===============================                                                                                       
Statements   : 99.45% ( 5220/5249 ), 52 ignored                                                                                                                        
Branches     : 98.2% ( 3977/4050 ), 24 ignored                                                                                                                         
Functions    : 100% ( 1123/1123 ), 18 ignored                                                                                                                          
Lines        : 99.45% ( 5220/5249 )                                                                                                                                    
================================================================================                                                                                       
                                                                                                                                                                       
                                                                                                                                                                       
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․   
                                                                                                                                                                       
  162 passing (164ms)   
```